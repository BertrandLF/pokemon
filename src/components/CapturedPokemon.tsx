import React from 'react'
import './styles/CapturedPokemon.css';
import Pokemon from './Pokemon';

interface CapturedPokemonProps {
  pokemon: Pokemon;
  handleOnClick: (pokemon: Pokemon) => void;
}

class CapturedPokemon extends React.Component<CapturedPokemonProps> {

  constructor (props: Readonly<CapturedPokemonProps>){
    super(props);
    this.getGif = this.getGif.bind(this);
    this.setNormalizedSrc = this.setNormalizedSrc.bind(this);
  }

  pokeGifUrl: string = 'http://pokestadium.com/sprites/xy/';

  getGif(pokemon: string) {
    return this.pokeGifUrl + pokemon + '.gif';
  }

  setNormalizedSrc(img: any){
    const normalizedPokemonName: string = this.props.pokemon.name.replace('-', '');
    img.target.src = this.pokeGifUrl + normalizedPokemonName + '.gif';
  }

  render() {
    const pokemon = this.props.pokemon;
    return (
      <div className='captured-pokemon' onClick={() => this.props.handleOnClick(pokemon)}>
        <img onError={this.setNormalizedSrc} src={this.getGif(pokemon.name)} className='sprite-image' alt="sprite" />
        <div className="pokemon-data">
          <div className="data-row">
            <p className="cell">ID:</p>
            <p className="cell">{pokemon.id}</p>
          </div>
          <div className="data-row">
            <p className="cell">Name:</p>
            <p className="cell">{pokemon.name}</p>
          </div>
          <div className="data-row">
            <p className="cell">Type:</p>
            <p className="cell">{pokemon.type}</p>
          </div>
        </div>
      </div>
    )
  }
}

export default CapturedPokemon;