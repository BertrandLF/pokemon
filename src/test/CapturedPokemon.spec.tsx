import React from 'react'
import CapturedPokemon from '../components/CapturedPokemon';
import { psyduck } from './PokemonFixture';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders pokemon details', () => {

  const onClick = () => {};
  const component = shallow(
    <CapturedPokemon
      pokemon={psyduck}
      handleOnClick={onClick}>
    </CapturedPokemon>
  );

  expect(toJson(component)).toMatchSnapshot();

});