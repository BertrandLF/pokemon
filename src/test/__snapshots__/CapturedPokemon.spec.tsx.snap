// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pokemon details 1`] = `
<div
  className="captured-pokemon"
  onClick={[Function]}
>
  <img
    alt="sprite"
    className="sprite-image"
    onError={[Function]}
    src="https://img.pokemondb.net/sprites/black-white/anim/normal/psyduck.gif"
  />
  <div
    className="pokemon-data"
  >
    <div
      className="data-row"
    >
      <p
        className="cell"
      >
        ID:
      </p>
      <p
        className="cell"
      >
        54
      </p>
    </div>
    <div
      className="data-row"
    >
      <p
        className="cell"
      >
        Name:
      </p>
      <p
        className="cell"
      >
        psyduck
      </p>
    </div>
    <div
      className="data-row"
    >
      <p
        className="cell"
      >
        Type:
      </p>
      <p
        className="cell"
      >
        water
      </p>
    </div>
  </div>
</div>
`;
